test/helpers/generators.go

```diff
@@ -12,17 +17,17 @@ import (
 	gwtypes "github.com/kong/gateway-operator/internal/types"
 	"github.com/kong/gateway-operator/pkg/consts"
 	"github.com/kong/gateway-operator/pkg/vars"
-	"github.com/kong/gateway-operator/test/helpers"
 )
 
 // GenerateGatewayClass generates the default GatewayClass to be used in tests
-func GenerateGatewayClass() *gatewayv1.GatewayClass {
+func GenerateGatewayClass(parametersRef *gatewayv1.ParametersReference) *gatewayv1.GatewayClass {
```

Comment:
While totally fine like this, what would you say for making the argument a variadic arg thus making it optional? `GenerateGatewayClass(nil)` at call sites look a bit odd to me hence the proposal

Reply:
yep yep that was the point of my comment above. Setting it as variadic for the sake of making it optional, would require returning an error as well (or accepting a `testing.T` if we want this function to fail in case the number of args > 1). I prefer having a pointer and passing nil in case no `parametersRef` is required, but this is just a matter of personal preference.